client/gui/CAnimation.cpp

```diff
@@ -736,6 +755,13 @@ void SDLImage::verticalFlip()
 	surf = flipped;
 }
 
+// Keep the original palette, in order to do color switching operation
+void SDLImage::savePalette()
+{
+	originalPalette = new SDL_Palette();

```

Comment:
Where does `originalPalette` being deleted? What if it has been saved already? Also you should use `SDL_AllocPalette`

Reply:
I will check this and update soon.